### PR TITLE
[4.0] Group action buttons for banner clients

### PR DESCRIPTION
**Pull request by chmst**
### Summary of Changes
Group buttons for status changes in a dropdown, see also  #22832

### Testing Instructions
Apply patch, mark one or more items in the list and perform status changes. Status changes must work as before.

bannerclients

### CODE REVIEW

File: administrator/components/com_banners/View/Clients/HtmlView.php [hunk 1]
```diff
@@ -84,37 +85,85 @@ public function display($tpl = null)
...
+		if ($canDo->get('core.edit.state') || ($this->state->get('filter.published') == -2 && $canDo->get('core.delete')))
 		{
-			ToolbarHelper::trash('clients.trash');
+			$dropdown = $toolbar->dropdownButton('status')
+				->text('JTOOLBAR_CHANGE_STATUS')
+				->toggleSplit(false)
+				->icon('fa fa-globe')
+				->buttonClass('btn btn-info')
+				->listCheck(true);
+
```
Comment (Quy) [inline administrator/components/com_banners/View/Clients/HtmlView.php]:
```suggestion

```

File: administrator/components/com_banners/View/Clients/HtmlView.php [hunk 2]
```diff
@@ -84,37 +85,85 @@ public function display($tpl = null)
 	protected function addToolbar()
 	{
 		$canDo = ContentHelper::getActions('com_banners');
-
+
```
Comment (Quy) [inline administrator/components/com_banners/View/Clients/HtmlView.php]: Remove tabs on lines 88 & 91.

File: administrator/components/com_banners/View/Clients/HtmlView.php (hunk 1 above)
Comment (Quy) [inline administrator/components/com_banners/View/Clients/HtmlView.php]: Remove tabs.

File: administrator/components/com_banners/View/Clients/HtmlView.php [hunk 3]
```diff
@@ -87,34 +88,46 @@ protected function addToolbar()
...
 
+		// Get the toolbar object instance
+		$toolbar = Toolbar::getInstance('toolbar');
+
 		if ($canDo->get('core.create'))
 		{
-			ToolbarHelper::addNew('client.add');
+			$toolbar->addNew('client.add');
 		}
+
```
Comment (Quy) [inline administrator/components/com_banners/View/Clients/HtmlView.php]: Remove tabs.

File: administrator/components/com_banners/View/Clients/HtmlView.php [hunk 4]
```diff
@@ -87,34 +88,46 @@ protected function addToolbar()
...
+			$toolbar->addNew('client.add');
 		}
+		
+		$dropdown = $toolbar->dropdownButton('status')
+				->text('JTOOLBAR_CHANGE_STATUS')
+				->toggleSplit(false)
+				->icon('fa fa-globe')
+				->buttonClass('btn btn-info')
+				->listCheck(true);
+
```
Comment (Quy) [inline administrator/components/com_banners/View/Clients/HtmlView.php]: Remove tab.

**Comment by Quy** [comment]
When clicking on the `Options` button, the following is displayed: 
`This component has no configuration options. `

File: administrator/components/com_banners/View/Clients/HtmlView.php [hunk 5]
```diff
@@ -87,34 +88,49 @@ protected function addToolbar()
...
 		elseif ($canDo->get('core.edit.state'))
 		{
-			ToolbarHelper::trash('clients.trash');
+			$childBar->trash('clients.trash')->listCheck(true);
 		}
 
 		if ($canDo->get('core.admin') || $canDo->get('core.options'))
 		{
-			ToolbarHelper::preferences('com_banners');
+			$toolbar->preferences('com_clients');
```
Comment (Quy) [inline administrator/components/com_banners/View/Clients/HtmlView.php]: When clicking on the `Options` button, the following is displayed: 
`This component has no configuration options. `

Comment (chmst) [reply administrator/components/com_banners/View/Clients/HtmlView.php]: Thank you! fixed.

**Comment by Hackwar** [comment]
I have tested this item :white_check_mark: successfully on d81b87073370d09139fb57ee2c9566edbf232701<br /><br />Tested this successfully<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23368">issues.joomla.org/tracker/joomla-cms/23368</a>.</sub>

File: administrator/components/com_banners/View/Clients/HtmlView.php [hunk 6]
```diff
@@ -87,34 +88,54 @@ protected function addToolbar()
...
+				->buttonClass('btn btn-info')
+				->listCheck(true);
+
+			$childBar = $dropdown->getChildToolbar();
+
+			$childBar->publish('clients.publish')->listCheck(true);
+			$childBar->unpublish('clients.unpublish')->listCheck(true);
+			$childBar->archive('clients.archive')->listCheck(true);
+
+			if ( $canDo->get('core.admin'))
```
Comment (Quy) [inline administrator/components/com_banners/View/Clients/HtmlView.php]:
```suggestion
			if ($canDo->get('core.admin'))
```

File: administrator/components/com_banners/View/Clients/HtmlView.php [hunk 7]
```diff
@@ -87,34 +88,54 @@ protected function addToolbar()
...
+				$childBar->checkin('clients.checkin')->listCheck(true);
+			}
+
+			if (!$this->state->get('filter.state') == -2)
+			{
+				$childBar->trash('clients.trash')->listCheck(true);
+			}
 		}
-
+
```
Comment (Quy) [inline administrator/components/com_banners/View/Clients/HtmlView.php]: Remove tabs.

File: administrator/components/com_banners/View/Clients/HtmlView.php (hunk 6 above)
Comment (Quy) [inline administrator/components/com_banners/View/Clients/HtmlView.php]:
```suggestion
			if ($canDo->get('core.admin'))
```

**Pull request merged**